### DISCUSSION TITLE
Add the announce block to only one section?

### DISCUSSION THREAD

**Discussion by StarfallProjects** [squidfunk/mkdocs-material#4027] [Q&A] [answered]
I am trying to add an announcement to just one docs section. It's a top-level section called "Embed" in the `nav`.

I assume I could make a layout for it that extends main, and then set that layout on every page of the section. But this feels clunky. So I am trying something like this, using the [section variable](https://www.mkdocs.org/dev-guide/themes/#section).

```
{% block announce %}
    {% if section and section.title == "Embed" %}
       n8n Embed requires a license. <a href="https://n8n-community.typeform.com/to/qOBgyWWA" target="_blank">Contact n8n</a> for more information.
    {% endif %}
{% endblock %}
```

However, this ends up with an empty announcement bar everywhere.

I have also tried this:

```
{% if section and section.title == "Embed" %}
{% block announce %}
       n8n Embed requires a license. <a href="https://n8n-community.typeform.com/to/qOBgyWWA" target="_blank">Contact n8n</a> for more information.
{% endblock %}
{% endif %}
```

Which ended up with the full announcement everywhere.

I suspect section is undefined (was getting that as an error before I added the `if section`)

Edit to add: Also, bizarrely, the block still gets added even when commented out.

**Comment by squidfunk** [comment, marked as answer]
Thanks for asking. The best idea is to use a dedicated page template which includes the `announce` block and [set this template for all pages of that section](https://squidfunk.github.io/mkdocs-material/reference/#setting-the-page-template), i.e. all pages that should include the announcement. Adding conditions to templates based on titles might be too unstable.

**Reply by StarfallProjects** [reply]
Ah ok! That's what I was hoping to avoid (having to edit a bunch of pages) Thanks for confirming there's no way round it 🙂

**Reply by squidfunk** [reply]
We could add the ability to add default metadata for specific folders, which would allow specifying a different template for all pages inside a folder (and nested folders). This is something I'm considering. You can open an issue with a feature request if you want.
